ms.go

```diff
@@ -0,0 +1,140 @@
+package ms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/coreos/go-oidc/v3/oidc"
+	"os"
+	"strings"
+)
+
+const rolesSeparator = "."
+const rolesKey = "roles"
+
+// Service defines basic parameters of a microservice for a client.
+type Service struct {
+	Host string
+	Port string
+}
+
+// BaseServiceServer contains a port on which the server listens for GRPC requests and provides JWT token verification.
+type BaseServiceServer struct {
+	Port       string
+	authIssuer string
+	verifier   *oidc.IDTokenVerifier
+}
+
+// Claims contains claims from a token and allows roles-checking for authorization.
+type Claims struct {
+	PreferredUsername string                         `json:"preferred_username"`
+	ResourceAccess    map[string]map[string][]string `json:"resource_access"`
+	RealmAccess       map[string][]string            `json:"realm_access"`
+}
+
+// GetRequiredEnv retrieves the value of the environment variable named by the key.
+// If the variable is not present in the environment an error is returned.
+func GetRequiredEnv(key string) (string, error) {
```

Comment:
The reason these functions are exported because they are reused across different microservices.
For example, if we define additional configuration parameters in environment variable like database host/driver, timeouts etc, we have to implement `GetRequiredEnv`/`GetOptionalEnv` ourselves again.

In the case of `FetchServiceParameters`, if some microservice wants to talk with another microservice, it has to know its host and port, that should be configurable.

Reply:
It might be time to use etcd for service registration and discovery. We can keep this for now, but this will need to be done this semester at some point.